qcodes/dataset/data_set_in_memory.py

```diff
@@ -0,0 +1,859 @@
+                name=name,
+                exp_id=exp_id,
+                exp_name=exp_name,
+                sample_name=sample_name,
+                guid=guid,
+                path_to_db=conn.path_to_dbfile,
+                run_timestamp_raw=None,
+                completed_timestamp_raw=None,
+                metadata=None,
+            )
+
+        return ds
+
+    @classmethod
+    def _load_from_netcdf(
+        cls, path: Union[Path, str], path_to_db: Optional[Union[Path, str]] = None
+    ) -> DataSetInMem:
+        """
+        Create a in memory dataset from a netcdf file.
+        The netcdf file is expected to contain a QCoDeS dataset that
+        has been exported using the QCoDeS netcdf export functions.
+
+
+        Args:
+            path: Path to the netcdf file to import.
+            path_to_db: Optional path to a database where this dataset may be
+             exported to. If not supplied the path can be given at export time
+             or the dataset exported to the default db as set in the QCoDeS config.
+
+        Returns:
+            The loaded dataset.
+        """
+        # in the code below flots and ints loaded from attributes are explicitly casted
+        # this is due to some older versions of qcodes writing them with a different backend
+        # reading them back results in a numpy array of one element
+
+        import xarray as xr
+
+        loaded_data = xr.load_dataset(path, engine="h5netcdf")
+
+        parent_dataset_links = str_to_links(
+            loaded_data.attrs.get("parent_dataset_links", "[]")
+        )
+        if path_to_db is not None:
+            path_to_db = str(path_to_db)
+
+        with contextlib.closing(
+            conn_from_dbpath_or_conn(conn=None, path_to_db=path_to_db)
+        ) as conn:
+            run_data = get_raw_run_attributes(conn, guid=loaded_data.guid)
+            path_to_db = conn.path_to_dbfile
+
+        if run_data is not None:
+            run_id = run_data["run_id"]
+            counter = run_data["counter"]
+        else:
+            run_id = int(loaded_data.captured_run_id)
+            counter = int(loaded_data.captured_counter)
+
+        path = str(path)
+        path = os.path.abspath(path)
+
+        export_info = ExportInfo.from_str(loaded_data.attrs.get("export_info", ""))
+        export_info.export_paths["nc"] = path
+        non_metadata = {
+            "run_timestamp_raw",
+            "completed_timestamp_raw",
+            "ds_name",
+            "exp_name",
+            "sample_name",
+            "export_info",
+            "parent_dataset_links",
+        }
+
+        metadata_keys = (
+            set(loaded_data.attrs.keys()) - set(RUNS_TABLE_COLUMNS) - non_metadata
+        )
+        metadata = {}
+        for key in metadata_keys:
+            data = loaded_data.attrs[key]
+            if isinstance(data, np.ndarray) and data.size == 1:
+                data = data[0]
+            metadata = {key: data}
```

Comment:
a typo, right? we want to populate metadata for all keys, right?)
```suggestion
            metadata[key] = data
```

Reply:
looks wrong. Check missing test coverage (at least 2 pieces of metadata in test